web/components/resources/FAQ.tsx

```diff
@@ -1,5 +1,242 @@
-import React from "react";
+import React, { useState, useEffect, Dispatch, SetStateAction } from 'react';
+import { doc, updateDoc, onSnapshot } from 'firebase/firestore';
+import UpChevron from '@components/Icons/UpChevron';
+import DownChevron from '@components/Icons/DownChevron';
+import TrashCan from '@components/Icons/TrashCan';
+import { db } from '@lib/firebase';
+import { AiFillWarning } from 'react-icons/ai';
+import { useRouter } from 'next/router';
 
-export default function FAQ() {
-  return <div>FAQ section here!</div>;
+type FAQEntry = {
+  question: string;
+  answer: string;
+  error?: boolean;
+};
+
+export default function FAQ(props: {
+  getChangesMade: () => boolean;
+  setChangesMade: Dispatch<SetStateAction<boolean>>;
+}) {
+  const [faqs, setFaqs] = useState<FAQEntry[]>();
+  const [userChanges, setUserChanges] = useState<FAQEntry[]>([]);
+  const router = useRouter();
+  useEffect(() => {
+    const unsub = onSnapshot(doc(db, 'resources', 'faq'), (doc) => {
+      setFaqs(doc.data()?.faqs || []);
+      setUserChanges(doc.data()?.faqs || []);
+    });
+
+    return unsub;
+  }, []);
+
+  useEffect(() => {
+    props.setChangesMade(JSON.stringify(userChanges) !== JSON.stringify(faqs));
```

Comment:
This will probably work in this case because the objects are the same, but beware that you'll see unexpected results in the future if you use this method (because JS objects are unordered).

Probably better would be to create a `compareChanges` function that checks if something has changed. Either way, I like how it's declarative instead of tracking a change in the dom.